kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -323,6 +321,131 @@ define(['common/jobMessages', 'common/jobs'], (JobMessages, Jobs) => {
         }
     }
 
+    /**
+     * A set of generic message handlers
```

Comment:
these run whenever the job manager is active; there will also be widget-specific handlers that do specific actions related to incoming messages and HTML components, e.g. the job status line updating itself or a row in the job status table reacting to a new status message.